hack/verify-test-featuregates.sh

```diff
@@ -36,19 +36,19 @@ if [[ -n "${direct_sets}" ]]; then
   echo "${direct_sets}" >&2
   echo >&2
   echo "Use this invocation instead:" >&2
-  echo "  defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, <value>)()" >&2
+  echo "  featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, <value>)" >&2
   echo >&2
   rc=1
 fi
 
 # find test files calling SetFeatureGateDuringTest and not calling the result
-missing_defers=$(git grep "\\.SetFeatureGateDuringTest" -- '*_test.go' | grep -E -v "defer .*\\)\\(\\)$") || true
+missing_defers=$(git grep "\\.SetFeatureGateDuringTest" -- '*_test.go' | grep -E "defer .*\\)\\(\\)$") || true
```

Comment:
this whole block can be dropped, right? this is now ensuring you *don't* call this with defer (which you can't do anyway?) and the warning output "Always make a deferred call" is no longer correct